src/components/ProductGridHero/ProductGridHero.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component, Fragment } from "react";
+import PropTypes from "prop-types";
+import { inject } from "mobx-react";
+import { withStyles } from "material-ui/styles";
+import Grid from "material-ui/Grid";
+import Fade from "material-ui/transitions/Fade";
+import Hidden from "material-ui/Hidden";
+import LoadingIcon from "mdi-material-ui/Loading";
+
+const styles = (theme) => ({
+  heroImg: {
+    width: "100%",
+    height: "325px",
+    objectFit: "cover"
+  },
+  productGridContainer: {
+    maxWidth: theme.grid.productGridMaxWidth,
+    marginLeft: "auto",
+    marginRight: "auto"
+  }
+});
+
+
+@withStyles(styles)
+@inject("uiStore")
+export default class ProductGridHero extends Component {
+  static propTypes = {
+    classes: PropTypes.object,
+    tag: PropTypes.object,
+    uiStore: PropTypes.object
+  };
+
+  static defaultProps = {
+    classes: {},
+    tag: {},
+    uiStore: {}
+  };
+
+  state = { hasImageLoaded: false };
+
+  onImageLoad = () => {
+    const { hasImageLoaded } = this.state;
+    if (hasImageLoaded) return;
+    this.setState({ hasImageLoaded: true });
+  };
+
+  buildImgUrl(imgPath) {
+    const { uiStore: { appConfig: { publicRuntimeConfig } } } = this.props;
+    return `${publicRuntimeConfig.externalAssetsUrl}${imgPath}`;
+  }
+
+  renderProductImage() {
+    const {
+      classes,
+      classes: { img, imgLoading, loadingIcon },
+      tag: { heroMediaUrl }
+    } = this.props;
+    const { hasImageLoaded } = this.state;
+
+    if (!heroMediaUrl) {
+      return null;
+    }
+
+    const picture = (
+      <picture>
+        <img
+          className={classes.heroImg}
+          src={heroMediaUrl}
+          alt=""
+          onLoad={this.onImageLoad}
+          ref={(image) => {
```

Comment:
should avoid defining a function in a render method. This "image loaded" component might be useful as a separate component.

Reply:
https://github.com/reactioncommerce/reaction-next-starterkit/issues/109